libp2p/stream/connection.nim

```diff
@@ -58,23 +58,28 @@ method initStream*(s: Connection) =
 
   procCall LPStream(s).initStream()
 
-  doAssert(isNil(s.timerTaskFut))
+  doAssert(s.timerTaskFut == nil)
 
   if s.timeout > 0.millis:
     trace "Monitoring for timeout", s, timeout = s.timeout
 
     s.timerTaskFut = s.timeoutMonitor()
-    if isNil(s.timeoutHandler):
-      s.timeoutHandler = proc(): Future[void] =
-        trace "Idle timeout expired, closing connection", s
-        s.close()
+    if s.timeoutHandler == nil:
```

Comment:
why `== nil` instead of `isNil`?

Reply:
https://github.com/status-im/nimbus-eth2/pull/4590#discussion_r1098580213